source/common/redis/conn_pool_impl.cc

```diff
@@ -125,6 +131,7 @@ void ClientImpl::onRespValue(RespValuePtr&& value) {
   ASSERT(!pending_requests_.empty());
   PendingRequest& request = pending_requests_.front();
   if (!request.canceled_) {
+    host_->outlierDetector().putHttpResponseCode(enumToInt(Http::Code::OK));
```

Comment:
@mattklein123 i could go either way. was originally thinking that in the case where there are more granular client-level timeouts it would be better to count cancellations as failures since you were unable to deliver downstream. the other causes of cancellations (client crash, etc) would probably be pretty small in number compared to that scenario.

in the future we will implement some method of providing more granular timeouts to envoy so the timeout would not have to be set for the long tail.

does that change your opinion at all? it's probably of negligible impact either way. i'm probably going to change it so that the stats make more sense. interested in your thoughts anyways.